src/functionaltests/java/com/ericsson/ei/subscriptions/crud/SubscriptionCRUDSteps.java

```diff
@@ -34,87 +32,105 @@
     private static final String SUBSCRIPTION_FILE_PATH = "src/functionaltests/resources/subscription_single.json";
     private static final String SUBSCRIPTION_UPDATED_FILE_PATH = "src/functionaltests/resources/subscription_single_updated.json";
 
-    @Autowired
-    private MockMvc mockMvc;
+    @LocalServerPort
+    private int applicationPort;
+    private String hostName = getHostName();
+    private HttpRequest httpRequest;
+    private ResponseEntity<String> response;
 
-    private MvcResult result;
     private ObjectMapper mapper = new ObjectMapper();
     private static JSONArray jsonArray = null;
 
     @Given("^The REST API \"([^\"]*)\" is up and running$")
     public void the_REST_API_is_up_and_running(String endPoint) throws Throwable {
-        result = mockMvc.perform(MockMvcRequestBuilders.get(endPoint).accept(MediaType.APPLICATION_JSON)).andReturn();
-        Assert.assertEquals(HttpStatus.OK.value(), result.getResponse().getStatus());
+        httpRequest = new HttpRequest(HttpMethod.GET);
+        httpRequest.setHost(hostName).setPort(applicationPort).setEndpoint(endPoint);
+        response = httpRequest.performRequest();
+        assertEquals(HttpStatus.OK, response.getStatusCode());
     }
 
     @When("^I make a POST request with valid \"([^\"]*)\" to the  subscription REST API \"([^\"]*)\"$")
-    public void i_make_a_POST_request_with_valid_to_the_subscription_REST_API(String arg1, String endPoint) throws Throwable {
+    public void i_make_a_POST_request_with_valid_to_the_subscription_REST_API(String arg1, String endPoint)
+            throws Throwable {
         String readFileToString = FileUtils.readFileToString(new File(SUBSCRIPTION_FILE_PATH), "UTF-8");
         jsonArray = new JSONArray(readFileToString);
-        result = mockMvc.perform(MockMvcRequestBuilders.post(endPoint).accept(MediaType.APPLICATION_JSON).content(jsonArray.toString())
-            .contentType(MediaType.APPLICATION_JSON)).andReturn();
+        httpRequest = new HttpRequest(HttpMethod.POST);
+        httpRequest.setHost(hostName).setPort(applicationPort).setEndpoint(endPoint)
+                .addHeader("content-type", "application/json").addHeader("Accept", "application/json")
+                .setBody(jsonArray.toString());
+        response = httpRequest.performRequest();
     }
 
-    @Then("^I get response code of (\\d+)$")
-    public void i_get_response_code_of(int statusCode) throws Throwable {
-        Assert.assertEquals(statusCode, result.getResponse().getStatus());
+    @Then("^I get response code of (\\d+)")
+    public void i_get_response_code_of(int statusCode) {
+        assertEquals(HttpStatus.valueOf(statusCode), response.getStatusCode());
```

Comment:
You can add failed message to the assertEquals(<message>, expected, actual)

Reply:
I can add a message but I'm not sure how to make it any clearer than the actual assert failure. The test is also set with pretty which makes it very clear at what scenario and step it failed.

Reply:
ok. If its clear what failed, then keep it as it is. I have seen assertions earlier which comapares to integers that it result is actual "0" is not "5",, but it was not clear what failed. My opinion is that test result should be as informative as possible so a user that don't that much about the test case can understand the failure in 1 or 2 seconds by just reading the failure message and result.

Reply:
In the number comparison I can add what failed if the assert fails but on the status code failure it's clear what was expected and at what scenario/step it failed.